core/trie/proof.go

```diff
@@ -40,14 +40,14 @@ func (b *Binary) String() string {
 
 type Edge struct {
 	Child *felt.Felt // child hash
-	Path  *Key       // path from parent to child
+	Path  *BitArray  // path from parent to child
 }
 
 func (e *Edge) Hash(hash hashFunc) *felt.Felt {
-	length := make([]byte, len(e.Path.bitset))
-	length[len(e.Path.bitset)-1] = e.Path.len
+	var length [32]byte
+	length[31] = e.Path.len
 	pathFelt := e.Path.Felt()
-	lengthFelt := new(felt.Felt).SetBytes(length)
+	lengthFelt := new(felt.Felt).SetBytes(length[:])
 	return new(felt.Felt).Add(hash(e.Child, &pathFelt), lengthFelt)
```

Comment:
This felt escapes to heap as well. I know this is not part of the PR so need to change it. I am making emphasis on this because some month back, the Cairo VM got twice as fast by removing this type of code. Just a note

Reply:
Why does `lengthFelt` escape to the heap? It's not returned anywhere

Reply:
Oh I see it now, you mean the returned `new(felt.Felt)`

Reply:
Yeah I'd avoid changing it in this PR, will add a TODO